pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/RoundRobinScheduler.java

```diff
@@ -58,12 +64,29 @@ public class RoundRobinScheduler implements OpChainScheduler {
   @VisibleForTesting
   final Set<MailboxIdentifier> _seenMail = new HashSet<>();
 
+  public RoundRobinScheduler() {
+    this(DEFAULT_RELEASE_TIMEOUT);
+  }
+
+  public RoundRobinScheduler(long releaseTimeout) {
+    this(releaseTimeout, System::currentTimeMillis);
+  }
+
+  public RoundRobinScheduler(long releaseTimeoutMs, Supplier<Long> ticker) {
+    _releaseTimeout = releaseTimeoutMs;
+    _ticker = ticker;
+  }
```

Comment:
this is a good idea since we might not always want to timeout based on sys time. To further strengthen can we do `_ticker = () -> System.currentTimeMillis() + _releaseTimeoutMs` ?  (e.g. putting _releaseTimeout as part of the _ticker?

one thinking: we might need to do "remaining time scheduling" in the future (this is what the V1 engine does)
  - e.g. broker computed the remaining time based on how many time it already spent on planning and dispatch. and the remaining time is used for timeouts, with this change I can simply replace `operatorChain::getRemainingScheduledTime` once we implemented it.

Reply:
we should not make the releaseTimeoutMS part of the ticker, the ticker is there for testing purposes so we can abstract away the system clock ticker (the `_ticker` is also used in `computeReady`).

if we want to also abstract away the release computation (e.g. pass in `shouldRelease`) we can do that and it can be `BiFunction<AvailableEntry, Long, Boolean>` where the first argument is the entry and the second is the current system time.

let's save that for a follow-up PR for when we want to make this more complicated

Reply:
ah. I see want you meant. sound good. sorry I misunderstood the purpose of this approach